cpp/src/IceBridge/IceBridge.cpp

```diff
@@ -523,28 +422,28 @@ BridgeI::ice_invoke_async(const AMD_Object_ice_invokePtr& cb,
             //
             target = target->ice_connectionId(Ice::generateUUID());
 
-            bc = new BridgeConnection(_adapter, target, current.con);
-            _connections.insert(make_pair(current.con, bc));
-            current.con->setCloseCallback(new CloseCallbackI(this));
+            bc = make_shared<BridgeConnection>(_adapter, target, current.con);
+            _connections.emplace(make_pair(current.con, bc));
```

Comment:
Perhaps use `{ current.con, bc }` instead of `make_pair`?

Reply:
The proper syntax is emplace(x, y), which constructs the pair automatically. Fixed.